common/fee/models.go

```diff
@@ -0,0 +1,102 @@
+package fee
+
+import (
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/v2/common/chains/label"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+var (
+	ErrBumpFeeExceedsLimit = errors.New("fee bump exceeds limit")
+	ErrBump                = errors.New("fee bump failed")
+	ErrConnectivity        = errors.New("transaction propagation issue: transactions are not being mined")
+)
+
+func IsBumpErr(err error) bool {
+	return err != nil && (errors.Is(err, ErrBumpFeeExceedsLimit) || errors.Is(err, ErrBump) || errors.Is(err, ErrConnectivity))
+}
+
+// CalculateFee computes the fee price and chain specific fee limit for a transaction.
+func CalculateFee(
+	feeLimit uint32,
+	maxFeePrice, defaultPrice, maxBumpPrice *big.Int,
+	bumpLimitMultiplier float32,
+) (feePrice *big.Int, chainSpecificFeeLimit uint32, err error) {
```

Comment:
Unused error return arg?
```suggestion
) (feePrice *big.Int, chainSpecificFeeLimit uint32) {
```

Reply:
thanks for bringing this up, i think it will be better to return a nil instead in the caller function